monitor/health/http.go

```diff
@@ -0,0 +1,131 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package health
+
+import (
+	"fmt"
+	"math/big"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/ChainSafe/ChainBridge/core"
+	msg "github.com/ChainSafe/ChainBridge/message"
+	log "github.com/ChainSafe/log15"
+)
+
+type HttpMetricServer struct {
+	port         int
+	core         *core.Core
+	blockheights map[msg.ChainId]BlockHeightInfo
+}
+
+type BlockHeightInfo struct {
+	height      *big.Int
+	lastUpdated time.Time
+}
+
+type httpMetricOptions struct {
+	port int
+	core *core.Core
+}
+
+//type chain struct {
+//	id int
+//}
+
+func newHTTPMetricServer(opts httpMetricOptions) *HttpMetricServer {
+	return &HttpMetricServer{
+		port:         opts.port,
+		core:         opts.core,
+		blockheights: make(map[msg.ChainId]BlockHeightInfo),
+	}
+}
+
+// Start starts the http metrics server
+func (s HttpMetricServer) Start() {
+	log.Info("Metrics server started", "port", s.port)
+
+	// Setup routes
+	http.HandleFunc("/health", s.healthStatus)
+
+	// Start http server
+	// TODO Push strconv to cli parser
+	err := http.ListenAndServe(":"+strconv.Itoa(s.port), nil)
+
+	if err == http.ErrServerClosed {
+		log.Info("Server is shutting down", err)
+	} else {
+		log.Error("Shutting down, server error: ", err)
+	}
+}
+
+// healthStatus is a catch-all update that grabs the latest updates on the running chains
+// It assumes that the configuration was set correctly, therefore the relevant chains are
+// only those that are in the core.Core registry.
+func (s HttpMetricServer) healthStatus(w http.ResponseWriter, r *http.Request) {
+
+	// Grab all chains
+	chains := s.core.GetChains()
+	requestTime := time.Now()
```

Comment:
Comparing the time of the requests to assert when the block was processed is pretty misleading. IMO timestamps should be established by the listener.